CMakeLists.txt

```diff
@@ -1,28 +1,24 @@
-cmake_minimum_required(VERSION 3.2.2)
-set(CMAKE_INSTALL_PREFIX ${CMAKE_BINARY_DIR} CACHE PATH
-  "Locaction for installation"
-  FORCE)
-set(CMAKE_CONFIGURATION_TYPES Release Debug)
-set(CMAKE_CONFIGURATION_TYPES "${CMAKE_CONFIGURATION_TYPES}" CACHE STRING "The needed configuration" FORCE)
-project(TemplateExternRepo)
-include (version.cmake)
-enable_testing()
-set(externpro_DIR "C:/Program Files/externpro 18.01.1-vc140-64")
-list(APPEND XP_MODULE_PATH ${externpro_DIR}/share/cmake)
-include(${externpro_DIR}/share/cmake/xpfunmac.cmake)
-xpCommonFlags()
+cmake_minimum_required(VERSION 3.21)
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+include(preproject)
+project(TemplateProject VERSION 0.0)
+set(externpro_REV 22.02)
+find_package(externpro REQUIRED)
+xpSetFlags()
+xpFindPkg(PKGS boost)
+xpSourceDirIgnore(cmake)
+xpSourceDirIgnore(Test)
 
-set(CMAKE_CXX_STANDARD 14)
-set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+# set(CMAKE_CXX_STANDARD 14)
+# set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
-if(CMAKE_COMPILER_IS_GNUCXX)
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -pedantic -Wextra")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Wall -pedantic -Wextra")
-else()
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
-endif(CMAKE_COMPILER_IS_GNUCXX)
-
-add_subdirectory(Shared)
-add_subdirectory(Executable)
+add_subdirectory(SolutionName)
 add_subdirectory(Playground)
+
+set(CPACK_PACKAGE_NAME "TEMPLATE")
+set(CPACK_PACKAGE_VENDOR "Landon Wilcox")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "${PACKAGE_NAME} Revision ${VERSION}")
+set(CPACK_PACKAGE_FILE_NAME "${PACKAGE_NAME}-${VERSION}")
+set(CPACK_PACKAGE_INSTALL_DIRECTORY "${PACKAGE_NAME}")
+set(CPACK_PACKAGE_VERSION ${VERSION})
+include(CPack)
```

Comment:
What type of package are you generating here? Does it default to zip?

Reply:
I have no idea 🤷‍♂️ I just copied this from your repo :D.

For the first project I'm going to base on this template, I don't actually care what windows does, but I assume I'll need to make an rpm on linux if more than just the single executable is made.
I was hoping that this will generate an rpm.